sdks/python/apache_beam/metrics/execution.py

```diff
@@ -0,0 +1,208 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Internal classes for Metrics API.
+
+The classes in this file keep shared state, and organize metrics information.
+
+Available classes:
+- MetricKey - Internal key for a metric.
+- MetricResult - Current status of a metric's updates/commits.
+- MetricsEnvironment - Keeps track of MetricsContainer and other metrics
+    information for every single execution working thread.
+- MetricsContainer - Holds the metrics of a single step and a single
+    unit-of-commit (bundle).
+"""
+from collections import defaultdict
+import threading
+
+from apache_beam.metrics.cells import CounterCell, DistributionCell
+
+
+class MetricKey(object):
+  """Metrics are internally keyed by the step name they associated with and
```

Comment:
Python style suggests a one-line docstring, followed by a longer description.  I suggest using: "Key used to identify instance of Metric.\n\n" followed by your existing comment.